kv/key.go

```diff
@@ -89,9 +91,26 @@ func (k Key) String() string {
 }
 
 // KeyRange represents a range where StartKey <= key < EndKey.
+// Hack: make the layout exactly the same with github.com/pingcap/kvproto/pkg/coprocessor.KeyRange
+// So we can avoid allocation of converting kv.KeyRange to coprocessor.KeyRange
+// Not defined as "type KeyRange = coprocessor.KeyRange" because their field name are different.
+// kv.KeyRange use StartKey,EndKey while coprocessor.KeyRange use Start,End
 type KeyRange struct {
 	StartKey Key
 	EndKey   Key
+
+	XXX_NoUnkeyedLiteral struct{}
+	XXX_unrecognized     []byte
+	XXX_sizecache        int32
+}
+
+func init() {
+	// kv.KeyRange and coprocessor.KeyRange should be the same layout, so we can exchange them.
+	var r1 KeyRange
+	var r2 coprocessor.KeyRange
+	if unsafe.Sizeof(r1) != unsafe.Sizeof(r2) {
+		panic("coprocessor protocol change?")
+	}
```

Comment:
I think we can make it a test. Also  use `require.Equal` to test if their default value is exactly same will be better.

Reply:
good point.

Reply:
Done